fvm/environment/value_store.go

```diff
@@ -153,7 +153,16 @@ func (store *valueStore) SetValue(
 		return errors.NewInvalidInternalStateAccessError(id, "modify")
 	}
 
-	err := store.meter.MeterComputation(
+	oldValue, err := store.GetValue(owner, keyBytes)
+	if err != nil {
+		return fmt.Errorf("get value failed: %w", err)
+	}
+	// no-op write
+	if bytes.Equal(oldValue, value) {
+		return nil
+	}
```

Comment:
This part is necessary because there we no-op writes to the slabs that were being counted as regular writes for meter invalidation.

I decided to not meter these writes as `ComputationKindSetValue`.

@fxamacker Any Idea why the slabs were being overwritten with the same value?

Reply:
@janezpodhostnik It is possible for slabs to be overwritten with the same value.  Atree doesn't require value to be comparable (only dictionary key is required to be comparable), so Atree can update the same value and mark slab as dirty.

For example, when Cadence updates Atree map with a value under existing key, atree compares key and overwrites corresponding element with the provided value.  Since value can't be compared, overwritting can happen for the same value.  

This aspect works the same for non-inlining version since we never require value to be comparable and therefore can't compare values before overwritten it.

If you provide a test case for the no-op writes, I can look into it more to confirm it fits into the general scenario I described.  This would be useful to confirm if what you encountered is the case I described.

Reply:
should they be metered as `ComputationKindGetValue` since we have to do an explicit get? or has that already been accounted for?

Reply:
That already happens by calling `store.GetValue`

Reply:
Just following up to thank Janez for tracking down test case on Aug 26 and good followup chat on Slack! :+1: 

Not our Slack chat, but this Discord thread maybe also useful:
https://discord.com/channels/613813861610684416/1108968095982293002/1276209950410150019